Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
     return primitiveValue.stringValue();
 }
 
-static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
+static CSSCounterStyleDescriptors::AdditiveSymbols translateAdditiveSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto pad = properties.getPropertyCSSValue(CSSPropertyPad).get();
-    if (!pad || !is<CSSValueList>(pad))
-        return CSSCounterStyleDescriptors::Pad();
+    auto additiveSymbols = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
+    if (!additiveSymbols)
+        return { };
+    auto* list = downcast<CSSValueList>(additiveSymbols.get());
+    if (!list)
+        return { };
+
+    CSSCounterStyleDescriptors::AdditiveSymbols result;
+    for (auto& additiveValue : *list) {
+        if (auto* additiveTuple = dynamicDowncast<CSSValuePair>(&additiveValue)) {
+            auto& weightValue = additiveTuple->first();
+            auto& symbolValue = additiveTuple->second();
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            result.constructAndAppend(symbolToString(&symbolValue), weight.value<unsigned>());
+        } else
+            return { };
+    }
+    return result;
+}
 
-    auto& list = downcast<CSSValueList>(*pad);
-    ASSERT(list.size() == 2);
-    auto length = downcast<CSSPrimitiveValue>(list.item(0));
+static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
+{
+    auto pad = properties.getPropertyCSSValue(CSSPropertyPad);
+    if (!pad)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(pad.get())))
+        return { };
```

Comment:
We should probably fix this in another patch, but this should use CSSValuePair
```suggestion
    CSSValueList* list = dynamicDowncast<CSSValueList>(pad.get());
    if (!list)
        return { };
```